### PR TITLE
[ARCTIC-462][AMS] Add implement for RewriteFiles to support Hive

### LINKED ISSUE #462: [Bug][AMS]: Major Optimize isn't support hive
### What happened?

Now, Hive can't read data after optimize.
Because Major commit use rewrite that isn't support Hive

### Affects Versions

0.3.2

### What engines are you seeing the problem on?

AMS

### How to reproduce

_No response_

### Relevant log output

_No response_

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by hzluting**
## Why are the changes needed?
fix #462 Add implement for RewriteFiles to support Hive

## Brief change log

  - *Add implement for RewriteFiles to support Hive*

## How was this patch tested?
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by baiyangtx — APPROVED** [review]

File: hive/src/test/java/com/netease/arctic/hive/op/TestRewriteFiles.java [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class TestRewriteFiles extends HiveTableTestBase {
+
+  @Test
+  public void testOverwriteUnkeyedPartitionTable() throws TException {
```
Comment (zhoujinsong) [inline hive/src/test/java/com/netease/arctic/hive/op/TestRewriteFiles.java:37]:
```suggestion
  public void testRewriteUnkeyedPartitionTable() throws TException {
```

File: hive/src/test/java/com/netease/arctic/hive/op/TestRewriteFiles.java [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+    Set<DataFile> initDataFiles = new HashSet<>(dataFileBuilder.buildList(initFiles));
+
+    OverwriteFiles overwriteFiles = table.newOverwrite();
+    initDataFiles.forEach(overwriteFiles::addFile);
+    overwriteFiles.commit();
+
+    applyUpdateHiveFiles(partitionAndLocations, s -> false, initFiles);
+    assertHivePartitionLocations(partitionAndLocations, table);
+
+    // ================== test overwrite all partition
```
Comment (zhoujinsong) [inline hive/src/test/java/com/netease/arctic/hive/op/TestRewriteFiles.java:57]:
```suggestion
    // ================== test rewrite all partition
```

File: hive/src/test/java/com/netease/arctic/hive/op/TestRewriteFiles.java [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+
+    OverwriteBaseFiles overwriteBaseFiles = table.newOverwriteBaseFiles();
+    initDataFiles.forEach(overwriteBaseFiles::addFile);
+    overwriteBaseFiles.withTransactionId(table.beginTransaction(""));
+    overwriteBaseFiles.commit();
+
+    applyUpdateHiveFiles(partitionAndLocations, s -> false, initFiles);
+    assertHivePartitionLocations(partitionAndLocations, table);
+
+    // ================== test overwrite all partition
```
Comment (zhoujinsong) [inline hive/src/test/java/com/netease/arctic/hive/op/TestRewriteFiles.java:94]:
```suggestion
    // ================== test rewrite all partition
```

File: hive/src/test/java/com/netease/arctic/hive/op/TestRewriteFiles.java [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+    Assert.assertEquals("table partition properties hive location is error",
+        hiveTable.getSd().getLocation(),
+        table.partitionProperty().get(TablePropertyUtil.EMPTY_STRUCT)
+            .get(HiveTableProperties.PARTITION_PROPERTIES_KEY_HIVE_LOCATION));
+    Assert.assertEquals("table partition properties transient_lastDdlTime is error",
+        hiveTable.getParameters().get("transient_lastDdlTime"),
+        table.partitionProperty().get(TablePropertyUtil.EMPTY_STRUCT)
+            .get(HiveTableProperties.PARTITION_PROPERTIES_KEY_TRANSIENT_TIME));
+
+    // ================== test overwrite table
```
Comment (zhoujinsong) [inline hive/src/test/java/com/netease/arctic/hive/op/TestRewriteFiles.java:200]:
```suggestion
    // ================== test rewrite table
```

**Review by zhoujinsong — CHANGES_REQUESTED** [review]
@hzluting I left some comments. PTAL

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Comment by hzluting** [comment]
> @hzluting I left some comments. PTAL

**Pull request merged**